### PR TITLE
Interactive-list

### PR DESCRIPTION
**Pull request by emmanuellekamwa**
In this project, I have:
- Created a new javascript file that contains methods related to the status update
- Added event listener to checkbox
- Updated items object's value for completed key upon user actions
- Stored the updated array in local storage

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -29,6 +37,20 @@ function renderList() {
...
+      renderList();
+    });
+  }
+}
+
+function checkStorage() {
+  if (localStorage.getItem('list')) {
+    const listData = JSON.parse(localStorage.getItem('list'));
+    list = listData;
+    renderList();
```
Comment (Lameck1) [inline src/index.js:53]: no-test-data


- [ ] As suggests in the screenshot above, this function is checking storage, yes, but not rendering the todo list. Let see why:
    - When loading the app, ideally the `localStorage` is still blank. But then, we are setting the `list` to `listData` which is a reference to a `currently dataless localStorage`.

# Suggestion

I think we need to set some data to` localStorage` first, something like this work for me when testing your app locally:

```JavaScript
function checkStorage() {
  if (!localStorage.getItem('list')) {
    localStorage.setItem('list', JSON.stringify(list));
  }
}

window.onload = () => {
  checkStorage();
  renderList();
};
```

## Note
When the app can't render tasks, there is really no way to properly test it for this milestone.

File: src/index.js [hunk 2]
```diff
@@ -29,6 +37,20 @@ function renderList() {
     task.appendChild(checkbox);
     task.appendChild(description);
     tasks.appendChild(task);
+    checkboxCheck(list, i, checkbox);
+    checkbox.addEventListener('click', () => {
+      change(list, i);
+      saveList();
+      renderList();
+    });
```
Comment (Lameck1) [inline src/index.js:45]: - [ ] The correct `eventListener` for this checkbox is `'change'` as per requirement. A `'click'` event also works but for this task it is preferably a `'change'` event.
- [ ] From my test, calling `renderList()` method inside this `eventListener` is buggy. Where is the problem? You are trying to do a recursive call to this method(calling `renderList()` within itself inside this `eventListener`). I suggest you refactor this` eventListener` outside of `renderList()` method.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @emmanuellekamwa, 

Good job so far!
Let's work on some issues that are still hampering our moving forward though.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bellom — APPROVED** [review]
## Hi @emmanuellekamwa 👏  

**Great work making all the necessary corrections from the previous review.**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**